Add single sign on documentation (#792)
sulu-io/sulu-docs: bundles/security/index.rst

from the role might be overridden by the permissions from this specific object
(which are handled by the previously mentioned ``AccessControlManager``).

> Single-Sign-On  Authentication

Reviewer: ```suggestion
Single-Sign-On Authentication
```

------------------------------

Sulu supports authentication via Single-Sign-On (SSO).
To enable it, the security configuration needs to be adjusted to allow SSO in the admin firewall.
This can be configured in the ``config/packages/security.yaml``:

.. code-block:: diff

    security:
        # ...

        firewalls:
            # ...
            admin:

                # ...
                logout:
                    path: sulu_admin.logout
   +            access_token:
   +                token_handler: sulu_security.single_sign_on_token_handler
   +                token_extractors: sulu_security.single_sign_on_token_extractor

    # ...
    sulu_security:
        checker:
            enabled: true
        password_policy:
            enabled: true
   +    single_sign_on:
   +        providers:
   +            'sulu.io':
   +                dsn: 'openid://%env(resolve:SULU_OPEN_ID_CLIENT_ID)%:%env(resolve:SULU_OPEN_ID_CLIENT_SECRET)%@%env(resolve:SULU_OPEN_ID_ENDPOINT)%'

>    +                user_role: 'USER'

Reviewer 2: While write the blog post I'm asking if we should name this `role` or `role_key`. The user role entity is a different one 🤔  /cc @chirimoya @Prokyonn

Reviewer 3: `default_role`, `default_role_key`?

Reviewer: I think `default_role_key` is the most accurate, I'd go with that

After adjusting the configuration and clearing the symfony cache,
you only see the ``username or email`` field when you try to login to the administration interface.

> When the user email matches the configured domain,
> the user is then redirected to the SSO provider to authenticate and after successful authentication,
> the user is redirected back to the administration interface.

Reviewer: ```suggestion
When the user email matches the configured domain,
the user is then redirected to the SSO provider to authenticate. After successful authentication, the system redirects the user back to the administration interface.
```

If the domain does not match the configured domain, the user is authenticated using the standard login form.
On password reset, when the domain matches, the user is also redirected to the SSO provider.

> Note: At the moment, only the OpenID protocol is supported for Single-Sign-On authentication in Sulu.

Reviewer: ```suggestion
.. note::

    At the moment, only the OpenID protocol is supported for Single-Sign-On authentication in Sulu.
```
